psc/src/main/java/com/pinterest/psc/metadata/client/PscMetadataClient.java

```diff
@@ -0,0 +1,172 @@
+        this.pscConfigurationInternal = new PscConfigurationInternal(
+                pscConfiguration,
+                PscConfigurationInternal.PSC_CLIENT_TYPE_METADATA
+        );
+        initialize();
+    }
+
+    private void initialize() {
+        creatorManager = new PscMetadataClientCreatorManager();
+        environment = pscConfigurationInternal.getEnvironment();
+    }
+
+    /**
+     * List all the {@link TopicRn}'s in the cluster.
+     *
+     * @param clusterUri
+     * @param timeout
+     * @param timeUnit
+     * @return the list of {@link TopicRn}'s in the cluster
+     * @throws ExecutionException
+     * @throws InterruptedException
+     * @throws TimeoutException
+     */
+    public List<TopicRn> listTopicRns(TopicUri clusterUri, long timeout, TimeUnit timeUnit) throws ExecutionException, InterruptedException, TimeoutException {
+        PscBackendMetadataClient backendMetadataClient = getBackendMetadataClient(clusterUri);
+        return backendMetadataClient.listTopicRns(timeout, timeUnit);
+    }
+
+    /**
+     * Describe the metadata for the given {@link TopicRn}'s in the cluster.
+     *
+     * @param clusterUri
+     * @param topicRns
+     * @param timeout
+     * @param timeUnit
+     * @return a map of {@link TopicRn} to {@link TopicRnMetadata}
+     * @throws ExecutionException
+     * @throws InterruptedException
+     * @throws TimeoutException
+     */
+    public Map<TopicRn, TopicRnMetadata> describeTopicRns(TopicUri clusterUri, Set<TopicRn> topicRns, long timeout, TimeUnit timeUnit) throws ExecutionException, InterruptedException, TimeoutException {
+        PscBackendMetadataClient backendMetadataClient = getBackendMetadataClient(clusterUri);
+        return backendMetadataClient.describeTopicRns(topicRns, timeout, timeUnit);
+    }
+
+    /**
+     * List the offsets for the given {@link TopicUriPartition}'s in the cluster.
+     *
+     * For each {@link TopicUriPartition}, the user can specify an {@link MetadataClientOption} to specify whether to
+     * get the earliest or latest offset.
+     *
+     * For example, if the user specifies {@link MetadataClientOption#OFFSET_SPEC_EARLIEST}, the client will return the earliest
+     * offset for the partition.
+     *
+     * @param clusterUri
+     * @param topicRnsAndOptions
+     * @param timeout
+     * @param timeUnit
+     * @return a map of {@link TopicUriPartition} to {@link MessageId}. The {@link MessageId} will contain the offset but
+     * not necessarily the timestamp (timestamp might be null or unset)
+     * @throws ExecutionException
+     * @throws InterruptedException
+     * @throws TimeoutException
+     */
+    public Map<TopicUriPartition, MessageId> listOffsets(TopicUri clusterUri, Map<TopicUriPartition, PscMetadataClient.MetadataClientOption> topicRnsAndOptions, long timeout, TimeUnit timeUnit) throws ExecutionException, InterruptedException, TimeoutException {
```

Comment:
Is it necessary to return a message id, if the goal is just to get the offset? There are existing consumer APIs for example that just work with a `long` as the offset.